From cc4ecce518d1d232820688595c2c2550907bbba6 Mon Sep 17 00:00:00 2001
From: Ankit Jain <ankitja@vmware.com>
Date: Wed, 25 Oct 2023 06:34:28 +0000
Subject: [PATCH] photon-os-installer: ostreeinstaller: Fix tmp mountpoint

- tmp mountpoint inside deployment is broken symlink
  causing failure in tmpfs mount command.
- Fixed it by creating symlink before doing mount

Change-Id: Iab1d585151616e4fbe3a9c4890996082f79eb3b8
Signed-off-by: Ankit Jain <ankitja@vmware.com>
---
 photon_installer/ostreeinstaller.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/photon_installer/ostreeinstaller.py b/photon_installer/ostreeinstaller.py
index c3e10bc..72a5fae 100755
--- a/photon_installer/ostreeinstaller.py
+++ b/photon_installer/ostreeinstaller.py
@@ -165,9 +165,9 @@ class OstreeInstaller(object):
         commit_number = self.get_commit_number(self.ostree_ref)
         self.do_systemd_tmpfiles_commands(commit_number)
 
-        self.run_lambdas([lambda: self.mount_devices_in_deployment(commit_number)], "mounting done")
         deployment = os.path.join(self.photon_root, f"ostree/deploy/photon/deploy/{commit_number}.0/")
         self.create_symlink_directory(deployment)
+        self.run_lambdas([lambda: self.mount_devices_in_deployment(commit_number)], "mounting done")
 
         if os.path.exists(loader1):
             cmd = []
-- 
2.23.1

